### PR TITLE
Add API Key support for Coinpaprika

### PR DESCRIPTION
**Pull request by cds95**
## Description

Allow NOPs to start the Coinpaprika adapter with an API KEY.  This is required as Coinpaprika will soon be severely limiting their free tier.

## Changes

- Support setting an `API_KEY` environment variable for Coinpaprika
- Support setting a `IS_TEST_MODE` environment variable for Coinpaprika.  This is required for now as this need to be passed in the header for the API key to have an effect whilst the Coinpaprika API is in the testing phase.  This can be removed later on once their API is in production.
- Add unit tests to verify the correct adapter configuration

## Steps to Test

1. Set `API_KEY` and `IS_TEST_MODE` and send a request to the adapter to verify that it is working.
2. Unset both environment variables and ensure that the adapter is still working.

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

**Comment by cds95** [comment]
We should deploy this to k8s, turn on the environment variables and soak the adapter for a while to make sure that we aren't hitting their tier limits.  I will do this once this change has been merged to develop.

File: packages/sources/coinpaprika/src/config.ts
Comment (austinborn) [inline packages/sources/coinpaprika/src/config.ts:16]: Can you add a TODO/shortcut ticket to remove this once the API is production-ready?

**Comment by justinkaseman** [comment]
Let's have this go out next week, hold off on merging for an hour or two while I get this week's release re-prepped.

File: packages/sources/coinpaprika/src/config.ts
Comment (cds95) [reply packages/sources/coinpaprika/src/config.ts:16]: Sure.  Created a ticket here https://app.shortcut.com/chainlinklabs/story/25884/remove-is-test-environment-variable-once-coinpaprika-api-goes-live-with-api-keys

**Comment by cds95** [comment]
Converting to draft PR to prevent this from accidentally getting merged in.

File: packages/sources/coinpaprika/src/config.ts
Comment (boxhock) [inline packages/sources/coinpaprika/src/config.ts:14]: We should use parseBool here, otherwise a false-y value could be treated as true here

**Review by boxhock — CHANGES_REQUESTED** [review]

**Review by boxhock — APPROVED** [review]

**Pull request merged**
